src/toil_scripts/gatk_germline/germline.py

```diff
@@ -1,688 +1,570 @@
+
+    get_bam = job.wrapJobFn(prepare_bam, uuid, url, config, rg_line=rg_line).encapsulate()
+
+    job.addChild(get_bam)
+    input_bam = get_bam.rv(0)
+    input_bai = get_bam.rv(1)
+
+    if config.preprocess:
+        preprocess = job.wrapJobFn(run_germline_preprocessing,
+                                   cores,
+                                   get_bam.rv(0),
+                                   get_bam.rv(1),
+                                   config.genome_fasta,
+                                   config.genome_dict,
+                                   config.genome_fai,
+                                   config.phase,
+                                   config.mills,
+                                   config.dbsnp,
+                                   xmx=config.xmx).encapsulate()
+        get_bam.addChild(preprocess)
+        if config.preprocess_only:
+            output_dir = os.path.join(config.output_dir, uuid)
+            filename = '{}.{}.bam'.format(uuid, config.suffix)
+            output_bam = job.wrapJobFn(upload_or_move_job,
+                                       filename,
+                                       preprocess.rv(0),
+                                       output_dir)
+            preprocess.addChild(output_bam)
+            return preprocess.rv(0), preprocess.rv(1)
+        else:
+            input_bam = preprocess.rv(0)
+            input_bai = preprocess.rv(1)
+
+    hc_disk = PromisedRequirement(lambda x: 2*x.size + human2bytes('5G'), input_bam)
+    haplotype_caller = job.wrapJobFn(gatk_haplotype_caller,
+                                     input_bam,
+                                     input_bai,
+                                     config,
+                                     memory=config.xmx,
+                                     disk=hc_disk)
+
+    get_bam.addFollowOn(haplotype_caller)
+    output_dir = os.path.join(config.output_dir, uuid)
+    vqsr_name = '{}.raw{}.gvcf'.format(uuid, config.suffix)
+    output_gvcf = job.wrapJobFn(upload_or_move_job,
+                                vqsr_name,
+                                haplotype_caller.rv(),
+                                output_dir)
+    haplotype_caller.addChild(output_gvcf)
+    return haplotype_caller.rv()
+
+
+def download_shared_files(job, config):
+    """
+    Downloads reference files shared by all samples in the pipeline
+
+    :param Namespace config: Pipeline configuration options
+    :return: Updated config with shared fileStoreIDS
+    :rtype: Namespace
+    """
+    job.fileStore.logToMaster('Downloading shared reference files')
+    references = {'genome_fasta'}
+    unrequired_references = {'genome_fai', 'genome_dict'}
+    references |= unrequired_references
+    if hasattr(config, 'ref'): config.genome_fasta = config.ref
+    if hasattr(config, 'fai'): config.genome_fai = config.fai
+    if hasattr(config, 'dict'): config.genome_dict = config.dict
```

Comment:
Not the preferred syntax for one-line if statements.

`config.genome_fasta = config.ref if hasattr(config, 'ref') else None`